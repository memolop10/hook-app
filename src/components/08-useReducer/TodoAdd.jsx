import React from 'react'
import useForm from '../../hooks/useForm';

const TodoAdd = ({ handleAddTodo }) => {

    const [ { description }, handleInputChange, reset ] = useForm({
        description:''
    });

    const handleSubmit = (e) => {
        e.preventDefault();

        if (description.trim().length <= 1) {
            return;
        }

        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false
        }

        handleAddTodo( newTodo )
        reset()
    }

    return (
        <>
          <h4>Agregar Todo</h4>
            <form onSubmit={ handleSubmit }>
                <input 
                    type="text" 
                    className="form-control"
                    name="description" 
                    placeholder="Aprender"
                    autoComplete="off"
                    value={ description }
                    onChange={ handleInputChange }
                />

                <button 
                    className="btn btn-block btn-dark mt-1"
                    type="submit"
                >
                    Agregar
                </button>
            </form>  
        </>
    )
}

export default TodoAdd
