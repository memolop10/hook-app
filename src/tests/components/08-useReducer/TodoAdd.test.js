import { shallow } from "enzyme";
import TodoAdd from "../../../components/08-useReducer/TodoAdd";


describe('Test in <TodoAdd />', () => {

    const handleAddTodo = jest.fn();

    const wrapper = shallow( 
        <TodoAdd 
           handleAddTodo={ handleAddTodo } 
        />
    )

    test('should render correctly ', () => {

        expect( wrapper ).toMatchSnapshot();
        
    })
    
    test('shouldnt call handleAddTodo ', () => {

        const formSubmit = wrapper.find('form').prop('onSubmit');

        formSubmit({ preventDefault(){} })
        
        expect( handleAddTodo ).toHaveBeenCalledTimes(0);
    })

    test('should call handleAddTodo ', () => {

       const value = 'Aprender VUE';
       wrapper.find('input').simulate('change',{
           target:{
               value,
               name:'description'
           }
       })

       const formSubmit = wrapper.find('form').prop('onSubmit');

       formSubmit({ preventDefault(){} })
       
       expect( handleAddTodo ).toHaveBeenCalledTimes(1);
       expect( handleAddTodo ).toHaveBeenCalledWith( expect.any(Object) ); // { }
       expect( handleAddTodo ).toHaveBeenCalledWith({
            id: expect.any(Number),
            desc: value,
            done: false
       })

       //despues de ejecutar todo esto el reset debe de limpiar el input
       expect( wrapper.find('input').prop('value') ).toBe('');
    })
});
