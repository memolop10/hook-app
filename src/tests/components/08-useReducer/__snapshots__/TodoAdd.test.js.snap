// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <TodoAdd /> should render correctly  1`] = `
<Fragment>
  <h4>
    Agregar Todo
  </h4>
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender"
      type="text"
      value=""
    />
    <button
      className="btn btn-block btn-dark mt-1"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
